### PR TITLE
Separators settings api

### PR DESCRIPTION
**Pull request by ManyTheFish**
# Summary

Allow the user to customize the list of separators used by Meilisearch to segment the documents and the queries by providing 2 new settings:
- `separatorTokens`: List of the tokens that should be considered separators
- `nonSeparatorTokens`: List of the tokens that should not be considered separators

# Changes

- add a new dedicated settings spec
- update the global settings spec
- update telemetry spec

## Misc

- [x] Update OpenAPI specification file
- [x] Update telemetry datapoints

### CODE REVIEW

File: text/0034-telemetry-policies.md [hunk 1]
```diff
@@ -60,6 +60,8 @@ The collected data is sent to [Segment](https://segment.com/). Segment is a plat
 | DistinctAttribute Updated | Occurs when distinct attribute setting is updated via `PUT` — `/indexes/:indexUid/settings/distinct-attribute`. |
 | DisplayedAttributes Updated | Occurs when displayed attributes are updated via `PUT` — `/indexes/:indexUid/settings/displayed-attributes`. |
 | StopWords Updated | Occurs when stop words are updated via `PUT` — `/indexes/:indexUid/settings/stop-words`. |
+| separatorTokens Updated | Occurs when separator tokens are updated via `PUT` — `/indexes/:indexUid/settings/separator-tokens`. |
+| nonSeparatorTokens Updated | Occurs when non separator tokens are updated via `PUT` — `/indexes/:indexUid/settings/non-separator-tokens`. |
```
Comment (macraig) [inline text/0034-telemetry-policies.md:64]: For consistency with the events above, should they be capitalized?

```suggestion
| NonSeparatorTokens Updated | Occurs when non separator tokens are updated via `PUT` — `/indexes/:indexUid/settings/non-separator-tokens`. |
```

File: text/0034-telemetry-policies.md [hunk 2]
```diff
@@ -60,6 +60,8 @@ The collected data is sent to [Segment](https://segment.com/). Segment is a plat
 | DistinctAttribute Updated | Occurs when distinct attribute setting is updated via `PUT` — `/indexes/:indexUid/settings/distinct-attribute`. |
 | DisplayedAttributes Updated | Occurs when displayed attributes are updated via `PUT` — `/indexes/:indexUid/settings/displayed-attributes`. |
 | StopWords Updated | Occurs when stop words are updated via `PUT` — `/indexes/:indexUid/settings/stop-words`. |
+| separatorTokens Updated | Occurs when separator tokens are updated via `PUT` — `/indexes/:indexUid/settings/separator-tokens`. |
```
Comment (macraig) [inline text/0034-telemetry-policies.md:63]:
```suggestion
| SeparatorTokens Updated | Occurs when separator tokens are updated via `PUT` — `/indexes/:indexUid/settings/separator-tokens`. |
```

**Review by macraig — DISMISSED** [review]

File: text/0123-separators-settings-api.md [hunk 3]
```diff
@@ -0,0 +1,164 @@
+# Stop Words Setting API
```
Comment (bidoubiwa) [inline text/0123-separators-settings-api.md:1]:
```suggestion
# Separators Setting API
```

File: text/0123-separators-settings-api.md [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+
+### 3.1. Explanations
+
+TBD
+
+#### 3.1.1. Usage Examples
+
+***Request payload `PUT`- `/indexes/articles/settings/separator-tokens`***
+```json
+["|", "/", "&sep"]
```
Comment (guimachiavelli) [inline text/0123-separators-settings-api.md:20]: Isn't `"/"` a separator by default? Using this list as a reference: https://docs.rs/charabia/0.8.3/src/charabia/separators.rs.html#16-62

Comment (ManyTheFish) [reply text/0123-separators-settings-api.md:20]: Yes it is

File: text/0123-separators-settings-api.md [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+This specification describes the `separatorTokens` and `nonSeparatorTokens` index setting API endpoints.
+
+## 2. Motivation
+N/A
+
+## 3. Functional Specification
+
+### 3.1. Explanations
+
+TBD
```
Comment (macraig) [inline text/0123-separators-settings-api.md:14]: Missing explanation

**Review by macraig — APPROVED** [review]

File: text/0123-separators-settings-api.md (hunk 5 above)
Comment (macraig) [reply text/0123-separators-settings-api.md:14]:
```suggestion
```

**Pull request merged**